### PR TITLE
Hometasks/hometask03

### PR DESCRIPTION
**Pull request by Frtn0717**

### CODE REVIEW

File: hometask03/add.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+function add() {
+  const args = [...arguments];
+  const result = {};
+
+  if(args.length === 0) {
+    throw new Error('Needs at least 1 argument');
+  } else if(args.length === 1) {
+    Object.assign(result, args[0]);
+    console.log(result);
```
Comment (ArtsiomFinenka) [inline hometask03/add.js:23]: Why do we need the second if in this statement? If you deleted this part, the last else will handle the single object

File: hometask03/add.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  y: 1,
+  z: 7,
+};
+
+function add() {
+  const args = [...arguments];
+  const result = {};
+
+  if(args.length === 0) {
+    throw new Error('Needs at least 1 argument');
```
Comment (ArtsiomFinenka) [inline hometask03/add.js:20]: instead of throwing an error let's return `null`

File: hometask03/add.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      const currentObject = args[i];
+
+      for(let key in currentObject) {
+        if(key in result) {
+          result[key] += currentObject[key];
+        } else {
+          result[key] = currentObject[key];
+        }
+      }
+    }
```
Comment (ArtsiomFinenka) [inline hometask03/add.js:37]: If you change this part using "reduce" method instead of for() then it will be pretty cool

File: hometask03/add.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+const obj4 = {
+  m: 2,
+  i: 7,
+  n: 3,
+  s: 5,
+  k: 0,
+}
+
+function findIntersect() {
```
Comment (ArtsiomFinenka) [inline hometask03/add.js:85]: The same comments like in add() function

File: hometask03/add.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+// findIntersect(obj2, obj3);
+
+// findIntersect(obj1, obj2, obj3);
+
+// findIntersect(obj1, obj2, obj3, obj4);
+
+// ------------------------
+
+function findIntersectInArr() {
```
Comment (ArtsiomFinenka) [inline hometask03/add.js:126]: The same comments like in add() function

**Review by ArtsiomFinenka — CHANGES_REQUESTED** [review]

**Review by ArtsiomFinenka — CHANGES_REQUESTED** [review]

File: hometask03/add.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          acc[key] += currItem[key];
+        }
+      });
+      Object.keys(acc).forEach(key => {
+        if (!currItem.hasOwnProperty(key)) {
+          delete acc[key];
+        }
+      });
+      return acc;
+    }, Object.assign({}, args.shift(0)));
```
Comment (ArtsiomFinenka) [inline hometask03/add.js:90]: `args.shift(0)` ???
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/shift

**Review by ArtsiomFinenka — APPROVED** [review]

File: hometask03/add.js (hunk 6 above)
Comment (Frtn0717) [reply hometask03/add.js]: My bad

**Pull request merged**